File: src/station_icp.cpp
#include "ros/ros.h"
#include <vector>
#include <iostream>
#include "std_msgs/String.h"
#include "sensor_msgs/LaserScan.h"

void scanCallback(const sensor_msgs::LaserScan::ConstPtr& scan_in){

  ROS_INFO("Scan Data Received");
  std::vector<float> range = scan_in->ranges;

}

void mapLayoutCallback(const std_msgs::String::ConstPtr& msg){

  ROS_INFO("Map Layout Data Received");
  std::string map_str = msg->data;

}

int main(int argc, char **argv){

  ros::init(argc, argv, "station_icp");
  ros::NodeHandle n;
  ros::Rate loop_rate(10);

  ros::Subscriber scan_sub = n.subscribe<sensor_msgs::LaserScan>("/scan", 10, scanCallback);
  ros::Subscriber mapLayout_sub = n.subscribe<std_msgs::String>("/map_layout", 10, mapLayoutCallback);

  while(ros::ok()){

    ros::spinOnce();

    loop_rate.sleep();
  
  }

  return 0;

}

File: scripts/station_icp.py
#!/usr/bin/env python

import rospy
import json
from std_msgs.msg import String
from sensor_msgs.msg import LaserScan

stations = list()

def scanCallback(data):
    pass
    #rospy.loginfo("Get Scan Data")

def mapLayoutCallback(data):

    map_json = json.loads(data.data)
    

    for item in map_json:
        print(str(item))

    print("Paths---------------------")
    for item in map_json["paths"]:
        print(str(item["cp1"]) + " " + str(item["cp2"]))

    print("\nJunction Count----------------")
    print(map_json["junction_count"])

    print("\nJunctionssss----------------")
    for item in map_json["junctions"]:
        print(str(item))

    print("\nStations---------------------")
    for item in map_json["stations"]:
        stations.append(item["name"])

    for station in stations:
        print(station)

def station_icp():
    
    rospy.init_node('station_icp', anonymous=False)
    rospy.Subscriber("/scan", LaserScan, scanCallback)
    rospy.Subscriber("/map_layout", String, mapLayoutCallback)
    rate = rospy.Rate(10)

    while not rospy.is_shutdown():
        rate.sleep()

if __name__ == '__main__':
    try:
        station_icp()
    except rospy.ROSInterruptException:
        pass

